library/src/main/kotlin/com/connectrpc/protocols/ConnectInterceptor.kt

```diff
@@ -93,7 +93,7 @@ internal class ConnectInterceptor(
                 trailers.putAll(response.headers.toTrailers())
                 trailers.putAll(response.trailers)
                 val responseHeaders =
-                    response.headers.filter { entry -> !entry.key.startsWith("trailer") }.toMutableMap()
+                    response.headers.filter { entry -> !entry.key.startsWith("trailer") }
```

Comment:
Noticed this pattern throughout - we were making headers mutable even though they weren't being mutated.